presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/MongoServer.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.mongodb;
+
+import org.testcontainers.containers.GenericContainer;
+
+import java.io.Closeable;
+
+public class MongoServer
+        implements Closeable
+{
+    private static final int MONGO_PORT = 27017;
+
+    private final GenericContainer<?> dockerContainer = new GenericContainer<>("mongo:3.2.0");
+
+    public MongoServer()
+    {
+        dockerContainer.withExposedPorts(MONGO_PORT)
```

Comment:
```suggestion
        dockerContainer
            .withExposedPorts(MONGO_PORT)
```

?

Reply:
`org.testcontainers.containers.Container#withExposedPorts` javadoc doesn't say -- is it binding 27017 on localhost, or _some port_ on `localhost`?

Is it binding on `127.0.0.1` or `0.0.0.0`?

cc @bsideup

Reply:
Neither. You should always use `dockerContainer.getContainerIpAddress()` instead of `localhost` and `dockerContainer.getMappedPort(27017)` to obtain the mapped port (always random)

Reply:
> (always random)

👍 

> Neither. You should always use `dockerContainer.getContainerIpAddress()` instead of `localhost`

i get it's not guaranteed to be `localhost` (depending on env configuration).

but, on my local env i see this

```
mongo:3.2.0                               0.0.0.0:32771->27017/tcp
quay.io/testcontainers/ryuk:0.2.3         0.0.0.0:32770->8080/tcp
```

-- shouldn't this bind to `127.0.0.1` instead?

(@ebyhr this is by no means a problem for this PR)

Reply:
We use "publish all" strategy (aka `-P` in CLI) and it (Docker) defaults to `0.0.0.0`.
Although we can't easily change it to `localhost`/`127.0.0.1` because there are some environments where Docker is deployed remotely

Reply:
> Although we can't easily change it to `localhost`/`127.0.0.1` because there are some environments where Docker is deployed remotely

can we use 127.0.0.1 when it's not?
eg how it's by default on Mac or Linux.

Reply:
Good idea, we will consider it! Could you please submit an issue to keep track of the progress?
https://github.com/testcontainers/testcontainers-java/issues/new

Reply:
https://github.com/testcontainers/testcontainers-java/issues/2133